Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm

```diff
@@ -506,6 +572,28 @@ static inline IntDegrees sensorOrientation(AVCaptureVideoOrientation videoOrient
 }
 ALLOW_DEPRECATED_DECLARATIONS_END
 
+void AVVideoCaptureSource::updateWhiteBalanceMode()
+{
+    auto mode = whiteBalanceModeFromMeteringMode(whiteBalanceMode());
+    ASSERT(mode);
+    if (!mode)
+        return;
+
+    NSError *error = nil;
+    [m_session beginConfiguration];
+    auto* device = this->device();
+    @try {
+        if ([device lockForConfiguration:&error])
+            device.whiteBalanceMode = mode.value();
```

Comment:
We are not doing so right now, but should we log the error if we are not able to lock?

Reply:
Good idea, fixed.